x/registry/keeper/msg_server_buy_alias.go

```diff
@@ -0,0 +1,57 @@
+package keeper
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/sonr-io/sonr/pkg/did"
+	"github.com/sonr-io/sonr/x/registry/types"
+)
+
+func (k msgServer) BuyAlias(goCtx context.Context, msg *types.MsgBuyAlias) (*types.MsgBuyAliasResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	if err := types.ValidateAlias(msg.GetName()); err != nil {
+		return nil, err
+	}
+
+	// Check if Alias exists
+	_, aliasIsFound := k.GetWhoIsFromAlias(ctx, msg.Name)
+	// If a name is found in store
+	if aliasIsFound {
+		return nil, sdkerrors.Wrap(types.ErrAliasUnavailable, "Name already has an owner")
+	}
+
+	// Get whois from Owner
+	// TODO: Implement Multisig for root level owner
+	whois, isFound := k.GetWhoIsFromOwner(ctx, msg.Creator)
+	if !isFound {
+		return nil, sdkerrors.Wrapf(types.ErrControllerNotFound, "creator %s", msg.Creator)
+	}
+
+	// Unmarshal DIDDoc from whois
+	doc := did.Document{}
+	err := doc.UnmarshalJSON(whois.GetDidDocument())
+	if err != nil {
+		return nil, sdkerrors.Wrap(types.ErrDidDocumentInvalid, "Failed to unmarshal DID document")
+	}
+
+	// Convert owner and buyer address strings to sdk.AccAddress
+	buyer, _ := sdk.AccAddressFromBech32(msg.Creator)
```

Comment:
Since the buyer comes from the message body we should first do checks to confirm the Creator can in fact perform this opperation